tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -3310,6 +3310,167 @@ llvm::Optional<Value> convertFakeQuantOp(PatternRewriter& rewriter,
       .getResult();
 }
 
+llvm::Optional<Value> convertMirrorPadCommon(PatternRewriter& rewriter,
+                                             Operation* op,
+                                             RankedTensorType output_type,
+                                             Value input, Value pad,
+                                             StringRef mode) {
+  RankedTensorType input_type = input.getType().dyn_cast<RankedTensorType>();
+  if (!input_type) {
+    (void)rewriter.notifyMatchFailure(op, "input type isn't a ranked tensor");
+    return llvm::None;
+  }
+
+  if ((mode != "SYMMETRIC") && (mode != "REFLECT")) {
+    (void)rewriter.notifyMatchFailure(op,
+                                      "mode isn't one of REFLECT or SYMMETRIC");
+    return llvm::None;
+  }
+
+  ElementsAttr pad_elems;
+  if (!matchPattern(pad, m_Constant(&pad_elems))) {
+    (void)rewriter.notifyMatchFailure(op, "fail to retrieve padding values");
+    return llvm::None;
+  }
+
+  // Indicate the value and shape of the tensor being processed in that
+  // dimension
+  Value current_tensor(input);
+  SmallVector<int64_t> current_dim_size;
```

Comment:
current_dim_size = to_vector(input_type.getShape()) ?

Reply:
It works. Nice function.